### PR TITLE
docs: add note about electron-updater

### PR DESCRIPTION
**Pull request by develar**

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
Repasting comment from other PR

> This kind of information should go in the update tutorial, that's where we moved a lot of it recently.

> Not sure it belongs in the docs though. The update documentation/tutorial is talking about using the electron autoUpdater API and I'm pretty sure that although electron-updater loosely wraps the API, it does a whole bunch of custom stuff that we can't be responsible for.

> Will defer to others but I don't think we should put something in the updater docs which we don't support and doesn't actually support our built in auto updating tech (It doesn't support Squirrel.Windows for instance).

**Comment by develar** [comment]
> The update documentation/tutorial is talking about using the electron autoUpdater API 

Since native autoUpdater has some drawbacks, we should point users to some possible solutions. i.e. no need to move all into the core — some things can be improved/implemented in the userland. In the same way as there are links to update servers. 

Tutorial is not only about core modules, but about "how to implement auto update in the Electron apps".

I didn't send PR earlier to stabilize/prove that electron-updater is really useful for users. As now electron-updater supports code signature validation, staged rollouts, static servers, progress events and so on on across all platfoms, it makes sense to add it here.

**Comment by zcbenz** [comment]
I'm unsure about this PR /cc @zeke

**Comment by develar** [comment]
@zcbenz now electron-updater supports differential updates for Windows Web installer (no delta file at all and corresponding configuration hell). In the October auto updates will be supported for Linux (https://github.com/electron-userland/electron-builder/issues/1138). So, lack of mention in the electron docs about complementary solutions will be confusing / strange for users.

File: docs/tutorial/updates.md [hunk 1]
```diff
@@ -94,3 +96,6 @@ autoUpdater.on('error', message => {
   console.error(message)
 })
 ```
+
+[electron-builder-lib]: https://github.com/electron-userland/electron-builder
+[electron-updater]: https://github.com/electron-userland/electron-builder/wiki/Auto-Update
```
Comment (TiagoDanin) [inline docs/tutorial/updates.md]: image

Use direct redirection to https://www.electron.build/auto-update

File: docs/tutorial/updates.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ but caches app updates on disk and supports private repositories.
 - [electron-release-server](https://github.com/ArekSredzki/electron-release-server) – 
 Provides a dashboard for handling releases
 
+When using [electron-builder][electron-builder-lib], you can also use userland [electron-updater][electron-updater] module. It allows you to avoid update server maintaining (Amazon S3, GitHub or any other static server are supported).
```
Comment (zeke) [inline docs/tutorial/updates.md]: I'm okay with landing this recommendation in the doc, but I would change the wording to this:

```
If your app is packaged with [electron-builder][electron-builder-lib] you can use the 
[electron-updater] module, which does not require a server and allows for updates 
from S3, GitHub, or any other static file host.
```

**Pull request merged**